src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/SplitFileService.cs

```diff
@@ -105,7 +162,7 @@ IImportStatusService importStatusService
 
                 // If no lines then don't create a batch or message unless it's the last one & there are zero
                 // lines in total in which case create a zero lines batch
-                if (table.Rows.Count == 0 && batchCount != numBatches || table.Rows.Count == 0 && messages.Count != 0)
+                if (table.Rows.Count == 0 && (batchCount != numBatches || batchFilesExist))
```

Comment:
Would be nice to add a little bit of extra detail to the comment to inform them what the `batchFilesExist` is I assume it's to do with the recovery process.

Reply:
I have added a comment next to the flag regarding EES-1608 - the ticket I raised to investigate all the existing checks around "no rows" batch files.